packages/itwin/grouping-mapping-widget/src/widget/hooks/useFetchAccessToken.ts

```diff
@@ -0,0 +1,22 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelApp } from "@itwin/core-frontend";
+import { useEffect, useState } from "react";
+
+const useFetchAccessToken = () => {
```

Comment:
heads up, this hook will provide a stale access token

Reply:
Good catch!

Reply:
might as well nuke the hook and just call getAccessToken directly

Reply:
Was thinking the same.

Reply:
Yeah, I'm not sure there is a way to do it without using a hook. Only places we use this are from the grids in imodel-browser. The prop requires an accessToken string. Small chance token expires while browsing for projects and iModel. I'll move it into the components that need it unless there is a better way.

Reply:
eh i guess could just leave the hook as is then, instead of duplicating the code in both SelectProject and IModel, but nbd; as long as you'll update it later